plugins/filter/filters.py

```diff
@@ -208,22 +208,46 @@ def listener_properties(self, listeners_dict, default_ssl_enabled,
+                final_dict['listener.name.' + listener_name + '.principal.builder.class'] =\
+                    'io.confluent.kafka.security.authenticator.OAuthKafkaPrincipalBuilder'
+
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and not oauth_enabled:
+                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + oauth_pem_path + '\";'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
                     'io.confluent.kafka.server.plugins.auth.token.TokenBearerValidatorCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.login.callback.handler.class'] =\
                     'io.confluent.kafka.server.plugins.auth.token.TokenBearerServerLoginCallbackHandler'
+
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and \
+                    oauth_enabled and rbac_enabled:
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
-                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + oauth_pem_path + '\";'
-                final_dict['listener.name.' + listener_name + '.principal.builder.class'] =\
-                    'io.confluent.kafka.security.authenticator.OAuthKafkaPrincipalBuilder'
+                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
+                    oauth_pem_path + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
+                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
+                    'io.confluent.kafka.server.plugins.auth.token.CompositeBearerValidatorCallbackHandler'
+                final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
+
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' \
+                    and oauth_enabled and not rbac_enabled:
+                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub="thePrincipalName";'
+                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
+                    'org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerValidatorCallbackHandler'
+                final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
```

Comment:
Can we not short these two if conditions?
if oauth_enabled:
  do all jwks url stuff
  
if rbac_enabled:
  add oauth_pem

Reply:
no, these are 3 different cases:
1. RBAC with ldap
2. RBAC with OAUTH
3. OAUTH without RBAC